netobserv/v1alpha1/flowcollector_types.go

```diff
@@ -0,0 +1,436 @@
+}
+
+// FlowCollectorEBPF defines a FlowCollector that uses eBPF to collect the flows information
+type FlowCollectorEBPF struct {
+	// Important: Run "make generate" to regenerate code after modifying this file
+
+	//+kubebuilder:default:="quay.io/netobserv/netobserv-ebpf-agent:main"
+	// Image is the NetObserv Agent image (including domain and tag)
+	Image string `json:"image,omitempty"`
+
+	//+kubebuilder:validation:Enum=IfNotPresent;Always;Never
+	//+kubebuilder:default:=IfNotPresent
+	// ImagePullPolicy is the Kubernetes pull policy for the image defined above
+	ImagePullPolicy string `json:"imagePullPolicy,omitempty"`
+
+	// Compute Resources required by this container.
+	// Cannot be updated.
+	// More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty" protobuf:"bytes,8,opt,name=resources"`
+
+	// Sampling is the sampling rate on the reporter. 100 means one flow on 100 is sent. 0 or 1 means disabled.
+	//+optional
+	Sampling int32 `json:"sampling,omitempty"`
+
+	// CacheActiveTimeout is the max period during which the reporter will aggregate flows before sending
+	//+kubebuilder:validation:Pattern:=^\d+(ns|ms|s|m)?$
+	//+kubebuilder:default:="5s"
+	CacheActiveTimeout string `json:"cacheActiveTimeout,omitempty"`
+
+	// CacheMaxFlows is the max number of flows in an aggregate; when reached, the reporter sends the flows
+	//+kubebuilder:validation:Minimum=1
+	//+kubebuilder:default:=1000
+	CacheMaxFlows int32 `json:"cacheMaxFlows,omitempty"`
+
+	// Interfaces contains the interface names from where flows will be collected. If empty, the agent
+	// will fetch all the interfaces in the system, excepting the ones listed in ExcludeInterfaces.
+	// If an entry is enclosed by slashes (e.g. `/br-/`), it will match as regular expression,
+	// otherwise it will be matched as a case-sensitive string.
+	//+optional
+	Interfaces []string `json:"interfaces,omitempty"`
+
+	// ExcludeInterfaces contains the interface names that will be excluded from flow tracing.
+	// If an entry is enclosed by slashes (e.g. `/br-/`), it will match as regular expression,
+	// otherwise it will be matched as a case-sensitive string.
+	//+kubebuilder:default=lo;
+	ExcludeInterfaces []string `json:"excludeInterfaces,omitempty"`
+
+	//+kubebuilder:validation:Enum=trace;debug;info;warn;error;fatal;panic
+	//+kubebuilder:default:=info
+	// LogLevel defines the log level for the NetObserv eBPF Agent
+	LogLevel string `json:"logLevel,omitempty"`
+
+	// Env allows passing custom environment variables to the NetObserv Agent. Useful for passing
+	// some very concrete performance-tuning options (e.g. GOGC, GOMAXPROCS) that shouldn't be
+	// publicly exposed as part of the FlowCollector descriptor, as they are only useful
+	// in edge debug/support scenarios.
+	//+optional
+	Env map[string]string `json:"env,omitempty"`
```

Comment:
We don't tend to use maps in Kube APIs. Normally we use a list with a map key (there's special semantics for this) as this in general is easier to consume for users merging the lists (think of multiple users managing individual entries in the list)